pkg/okteto/action_test.go

```diff
@@ -212,16 +190,34 @@ func TestWaitForActionToFinish(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			timerChannel := make(chan time.Time, 1)
+			tickerChannel := make(chan time.Time, 1)
 			pc := pipelineClient{
 				client: tc.cfg.client,
-				tickerWithTimeoutProvider: func(_, _ time.Duration) oktetoTime.TickerWithTimeoutInterface {
-					return &fakeTickerWithTimeout{
-						numberOfCallsBeforeTimeout: len(tc.cfg.client.queryResults),
-						tickerC:                    make(chan time.Time, 1),
-						timeoutC:                   make(chan time.Time, 1),
+				provideTicker: func(_ time.Duration) *time.Ticker {
+					return &time.Ticker{
+						C: tickerChannel,
+					}
+				},
+				provideTimer: func(d time.Duration) *time.Timer {
+					return &time.Timer{
+						C: timerChannel,
 					}
 				},
 			}
+
+			go func() {
+				for i := 0; i < len(tc.cfg.client.queryResults); i++ {
+					tickerChannel <- time.Now()
+					// Wait until the ticker has been read
+					for {
+						if len(tickerChannel) == 0 {
+							break
+						}
+					}
```

Comment:
I think this is not needed since _"By default, sends and receives block until the other side is ready. This allows goroutines to synchronize without explicit locks or condition variables."_